auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/AdminEditUserController.kt

```diff
@@ -35,22 +35,18 @@ class AdminEditUserController(
             arrayOf(DeltaSystemRole.ADMIN), call
         )
 
-        val cn = call.request.queryParameters["userCn"]!!
-        logger.atInfo().log("Request to update user {}", cn)
-
-        val (userToUpdate, userToUpdateRoles) = try {
-            userLookupService.lookupUserByCNAndLoadRoles(cn)
-        } catch (e: NameNotFoundException) {
-            throw ApiError(
-                HttpStatusCode.BadRequest,
-                "no_existing_user",
-                "Attempting to update a user that doesn't exist",
-            )
-        }
+        val userToUpdate = getUserFromCallParameters( // TODO DT-1022 - just get GUID
+            call.request.queryParameters,
+            userLookupService,
+            "Something went wrong, please try again",
+            "update_user_as_admin"
+        )
+        val userToUpdateRoles = userLookupService.loadUserRoles(userToUpdate).roles // TODO DT-1022 - use lookupUserByGUIDAndLoadRoles
 
+        logger.atInfo().log("Updating user with GUID ${userToUpdate.getGUID()}")
         val updatedDeltaUserDetailsRequest = call.receive<DeltaUserDetailsRequest>()
 
-        if (updatedDeltaUserDetailsRequest.id.replace("@", "!") != cn) throw ApiError(
+        if (updatedDeltaUserDetailsRequest.id.replace("@", "!") != userToUpdate.cn) throw ApiError(
```

Comment:
Should this use `LDAPConfig.emailToCN`?

Reply:
Yeah, that wasn't something I changed in this PR but I've now checked for other uses and changed it in 4 places total :)